Sources/ATTNAPI.m

```diff
@@ -42,6 +95,95 @@ - (void)sendUserIdentity:(ATTNUserIdentity *)userIdentity domain:(NSString *)dom
     }];
 }
 
+- (void)sendEvent:(id<ATTNEvent>)event userIdentity:(ATTNUserIdentity*)userIdentity domain:(NSString*) domain {
+    [self getGeoAdjustedDomain:domain completionHandler:^(NSString* geoAdjustedDomain, NSError* error) {
+        if (error) {
+            NSLog(@"Error sending user identity: '%@'.", error);
+            
+        }
+        
+        [self sendEventInternal:event userIdentity:userIdentity domain:domain];
+    }];
+}
+
+- (void)sendEventInternal:(id<ATTNEvent>)event userIdentity:(ATTNUserIdentity*)userIdentity domain:(NSString*) domain {
+    // slice up the Event into individual EventRequests
+    NSArray<EventRequest *>* requests = [self convertEventToRequests:event];
+    
+    for (EventRequest* request in requests) {
+        [self sendEventInternalForRequest:request userIdentity:userIdentity domain:domain];
+    }
+}
+
+- (void)sendEventInternalForRequest:(EventRequest*)request userIdentity: (ATTNUserIdentity*)userIdentity domain:(NSString*) domain {
+    NSURL* url = [self constructEventUrlComponentsForEventRequest:request userIdentity:userIdentity domain:domain].URL;
+
+    NSURLSessionDataTask* task = [_urlSession dataTaskWithURL:url completionHandler:^ void (NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (error) {
+            NSLog(@"Error sending for event '%@'. Error: '%@'", request.eventNameAbbreviation, [error description]);
+            return;
+        }
+        
+        // The response is an HTTP response because the URL had an HTTPS scheme
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse*) response;
+        if ([httpResponse statusCode] != 200) {
+            NSLog(@"Error sending the event '%@'. Incorrect status code: '%ld'", request.eventNameAbbreviation, (long)[httpResponse statusCode]);
+            return;
+        }
+        
+        NSLog(@"%@", [NSString stringWithFormat:@"Successfully sent event of type '%@'", request.eventNameAbbreviation]);
+    }];
+    
+    [task resume];
+}
+
+// A single event can create multiple requests to our servers. This method converts an event to multiple request objects.
+- (NSArray<EventRequest *>*)convertEventToRequests:(id<ATTNEvent>)event {
```

Comment:
we're gonna want an OrderConfirmed event produced as well

Reply:
Do we want that now?